run-go-mod-tidy.sh

```diff
@@ -1,13 +1,12 @@
 #!/usr/bin/env bash
 
-if go mod tidy -v 2>&1 | grep -q 'updates to go.mod needed'; then
-    exit 1
+go mod tidy -v $@
+if [ $? -ne 0 ]; then
```

Comment:
Looking at https://github.com/golang/go/blob/master/src/cmd/go/internal/modcmd/tidy.go, it seems like `go mod tidy` does not exist with anything except for `0`.  

I think we still need `| grep -q '...'` to check if `go mod tidy` found problems or not

Reply:
I'm not familiar with these code, but after a simple browse, `go mod tidy` exit by panic or os.Exit in functions?

In order to verify, we can change any existing dependency to error address, such `github.com/j2gg0s/otsql @xxx` -> `github.com/j2gg1s/otsql @xxx`, and run `go mod tidy`.

Reply:
Hmm, I think you might be right. Seems like maybe that error does result in a `os.Exit` from `Fatalf` here: https://github.com/golang/go/blob/122a22e0e9eba7fe712030d429fc4bcf6f447f5e/src/cmd/go/internal/modload/load.go#L443-L445

Have you been able to test it? I can't seem to get it to print that error.

Reply:
I don't known when `go mod tidy` will exit by this code, so use case is difficult to construct.

Answer in this issue also proves that go mod tidy will exit no zero for error.
https://github.com/golang/go/issues/27005